File: javascript/resistor-color/resistor-color.js
let colorMap = new Map();

colorMap.set('black',0);
colorMap.set('brown',1);
colorMap.set('red',2);
colorMap.set('orange',3);
colorMap.set('yellow',4);
colorMap.set('green',5);
colorMap.set('blue',6);
colorMap.set('violet',7);
colorMap.set('grey',8);
colorMap.set('white',9);

export const colorCode = (color) => {
  return colorMap.get(color);
};

export const COLORS = Array.from(colorMap.keys());

File: csharp/resistor-color/ResistorColor.cs
using System;

public static class ResistorColor
{
    public static string[] ColorMap => new[] { "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white" };

    public static int ColorCode(string color) =>  Array.FindIndex(ColorMap, x => x.Equals(color));
    public static string[] Colors() => ColorMap;
}